yorkie-team/yorkie-js-sdk
# Fix event-related tests to be deterministic

#### What this PR does / why we need it?

This is a follow-up to the work in https://github.com/yorkie-team/yorkie-js-sdk/pull/599/commits/b2eb6d45ef35e9b7300d8ac653538eda80479566. I modified the event-related tests to ensure deterministic behavior.

There are two ways to test events:
- When the count and order of events are known:
  - Use `EventCollector.waitAndVerifyNthEvent()` to wait for the nth event to occur and verify it.
- When the count and order of events are unclear, but we need to check if certain events are received:
  - Use `EventCollector.waitFor()` to verify the occurrence of specific events.

#### Any background context you want to provide?


#### What are the relevant tickets?

Fixes #

### Checklist
